### PR TITLE
[bridge 7/n] add BridgeClient

### PR DESCRIPTION
**Pull request by longbowlu**
## Description 

This PR:
1. adds BridgeClient, the component that talks to BridgeServer to get signatures
2. update `verify_signed_bridge_action` to make sure action and signer are expected
3. move `server.rs` and `handler.rs` to folder `server`

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-bridge/src/server/mod.rs [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+pub mod handler;
+
+#[cfg(test)]
+pub(crate) mod mock_handler;
+
+pub const APPLICATION_JSON: &str = "application/json";
+
+// Important: the paths need to match the ones in bridge_client.rs
+pub const ETH_TO_SUI_TX_PATH: &str = "/sign/bridge_tx/eth/sui/:tx_hash/:event_index";
+pub const SUI_TO_ETH_TX_PATH: &str = "/sign/bridge_tx/sui/eth/:tx_digest/:event_index";
```
Comment (longbowlu) [inline crates/sui-bridge/src/server/mod.rs:27]: for reviewer: this file was moved from `server.rs`, these 3 lines. No need to look at other parts other than these 3 lines

File: crates/sui-bridge/src/crypto.rs [hunk 2]
```diff
@@ -63,27 +66,41 @@ impl BridgeAuthoritySignInfo {
...
 }
 
-pub fn verify_signed_bridge_event(
-    e: SignedBridgeAction,
+/// Verifies a SignedBridgeAction (response from bridge authority to bridge client)
+/// represents the right BridgeAction, and is signed by the right authority.
+pub fn verify_signed_bridge_action(
+    expected_action: &BridgeAction,
+    signed_action: SignedBridgeAction,
+    expected_signer: &BridgeAuthorityPublicKeyBytes,
```
Comment (benr-ml) [inline crates/sui-bridge/src/crypto.rs:74]: why is this needed, isn't the committee enough?

Comment (longbowlu) [reply crates/sui-bridge/src/crypto.rs:74]: this function is called in BridgeClient::request_sign_bridge_action, where client requests a server to sign an action. Passing the `expected_signer` to match `&sig.authority_pub_key_bytes()` is to make sure the server returns its own signature (e.g. not another authority's, which may also be in the committee)

File: crates/sui-bridge/src/types.rs [hunk 3]
```diff
@@ -102,7 +108,7 @@ pub enum TokenId {
     USDT = 4,
 }
 
-#[derive(Clone, Debug, Serialize, Deserialize)]
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
```
Comment (mwtian) [inline crates/sui-bridge/src/types.rs:111]: For the TokenId enum above, do we want to make it more extensible?

File: crates/sui-bridge/src/server/handler.rs [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+use crate::eth_client::EthClient;
+use crate::sui_client::SuiClient;
+use crate::types::SignedBridgeAction;
+use async_trait::async_trait;
+use axum::Json;
+use sui_sdk::SuiClient as SuiSdkClient;
+
+#[async_trait]
+pub trait BridgeRequestHandlerTrait {
+    async fn handle_eth_tx_hash(
```
Comment (mwtian) [inline crates/sui-bridge/src/server/handler.rs:14]: Comment on what each handler does?

**Review by mwtian — APPROVED** [review]

**Review by benr-ml — APPROVED** [review]

File: crates/sui-bridge/src/types.rs (hunk 3 above)
Comment (longbowlu) [reply crates/sui-bridge/src/types.rs:111]: how to make it more extensible?

File: crates/sui-bridge/src/server/handler.rs (hunk 4 above)
Comment (longbowlu) [reply crates/sui-bridge/src/server/handler.rs:14]: will do in a later PR

Comment (longbowlu) [reply crates/sui-bridge/src/server/handler.rs:14]: https://github.com/MystenLabs/sui/pull/15290/files#diff-1489047ec8e9221e8c8c0990e62aa663f06fa60dbbece5df6d1fd7e4a4ebac99R14-R24

File: crates/sui-bridge/src/types.rs (hunk 3 above)
Comment (mwtian) [reply crates/sui-bridge/src/types.rs:111]: For TokenID, does the below work?
```
struct TokenID {
    name: String,
    native_chain: String,
}
```
Basically I wonder if the TokenID has to be an enum.

**Comment by lxfind** [comment]
Do we want to add authentication support (e.g. shared secrets among servers/client) on these requests, or would they be open to anyone to query?

File: crates/sui-bridge/src/types.rs (hunk 3 above)
Comment (longbowlu) [reply crates/sui-bridge/src/types.rs:111]: I see. It's made an enum so we can easily present it as an u8. We don't need the source chain info in v1 (and probably not until we support more chains)

**Comment by longbowlu** [comment]
> Do we want to add authentication support (e.g. shared secrets among servers/client) on these requests, or would they be open to anyone to query?

open to anyone + caching + ddos protection (e.g. rate limiting infra in the front)
